src/mpl/configure.ac

```diff
@@ -206,63 +191,58 @@ if test "$enable_fast_alwaysinline" = "yes"; then
     AC_DEFINE(ENABLE_ALWAYS_INLINE,1,[Define if force compiler to always inline functions with MPL_STATIC_INLINE_PREFIX|SUFFIX])
 fi
 
+AC_CANONICAL_HOST
+
+case "$host_cpu" in
+  i*86 | x86_64)
+    build_for_x86="yes"
+    ;;
+esac
+AM_CONDITIONAL([MPL_BUILD_FOR_X86], [test "x$build_for_x86" = "xyes"])
+
 # check if compiler supports avx, avx512f
 # also check if the building machine can run them if not force enabled
-AC_CACHE_CHECK([whether -mavx is supported], pac_cv_found_avx,
-              [PAC_C_CHECK_COMPILER_OPTION([-mavx],pac_cv_found_avx=yes,pac_cv_found_avx=no)],
+AC_CACHE_CHECK([whether -mavx2 is supported], pac_cv_found_avx,
```

Comment:
is this change to avx2 intended?

Reply:
Yes. I looked up on Intel's doc. Turns out the non-temporal load instruction that ZE code use is part of AVX2 not AVX. Since AVX2 is a superset of AVX, switching to it seems cleaner.